core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.component.ts

```diff
@@ -134,12 +134,19 @@ export class TopicsAndSkillsDashboardPageComponent {
     this.directiveSubscriptions.unsubscribe();
   }
 
-  generateNumbersTillRange(range: number): number[] {
-    let arr: number[] = [];
-    for (let i = 0; i < range; i++) {
-      arr.push(i);
+  getPages(current: number, total: number): number[] {
```

Comment:
Make the numbers here into named constants to explain what they represent (e.g. what is the 6 for), and add a comment to explain what -1 in the return value symbolizes. Explain, in comments, why you are doing each of the cases, so it's easy for another developer to understand the logic in the future.

Reply:
@seanlip Have added the comments

Reply:
Suggest using: currentPageNumber, totalNumberOfPages (since, unlike e.g. "current", those are clearly numbers) and changing the method name to getPagesToDisplay() or getPagesToDisplayInNavigation().